File: juste_prix.lua
-- Lua version améliorée

function juste_prix()
    math.randomseed(os.time())
    local random_price = math.random(1, 100)
    print(random_price)
    local tentatives = 0
    print("Devinez un nombre entre 1 et 100 !")
    while true do
        tentatives = tentatives + 1
        print("Tentative " .. tentatives .. " - Quel prix")
        local user_price_str = io.read()
        local user_price = tonumber(user_price_str)
        if user_price == nil then
            print("Erreur : Veuillez entrer un nombre valide ! (Cette tentative compte)")
        elseif user_price == random_price then
            print("Bien joué ! Vous avez trouvé en " .. tentatives .. " tentative(s) !")
            break
        elseif user_price < random_price then
            print("C'est plus")
        else
            print("C'est moins")
        end
    end
end
juste_prix()

File: juste_prix.java
// Java version améliorée

import java.util.Scanner;
import java.util.InputMismatchException;

class Main {
    public static void main(String[] args) {
        int random_price = (int)(Math.random() * 100) + 1;
        System.out.println(random_price);
        Scanner scanner = new Scanner(System.in);
        int tentatives = 0;
        System.out.println("Devinez un nombre entre 1 et 100 !");
        while (true) {
            tentatives++;
            System.out.print("Tentative " + tentatives + " - Quel prix : ");
            try {
                int price = scanner.nextInt();
                if (price == random_price) {
                    System.out.println("Bien joué ! Vous avez trouvé en " + tentatives + " tentative(s) !");
                    break;
                } else if (price < random_price) {
                    System.out.println("Ca vaut plus");
                } else {
                    System.out.println("Ca vaut moins");
                }
            } catch (InputMismatchException e) {
                System.out.println("Erreur : Veuillez entrer un nombre valide ! (Cette tentative compte)");
                scanner.nextLine();
            }
        }
        scanner.close();
    }
}